apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -184,63 +281,31 @@ defmodule Aecore.Peers.Sync do
   defp build_chain(state, block, chain) do
     has_parent_block_in_state = Map.has_key?(state, block.header.prev_hash)
     has_parent_in_chain = Chain.has_block?(block.header.prev_hash)
-    cond do
-      has_parent_block_in_state ->
-        build_chain(state, state[block.header.prev_hash], [block | chain])
-      has_parent_in_chain ->
-        [block | chain]
-      true ->
-        []
+    block_header_hash = BlockValidation.block_header_hash(block.header)
+    if(!Chain.has_block?(block_header_hash)) do
```

Comment:
Please remove parenthesis around if. (To find such errors in future you can use `mix credo -a` it should point this out.)